### ISSUE TITLE
[BUG] SquaredDistrLoss uses log_pdf instead of pdf

### ISSUE THREAD

**Issue by MayankSharma-2812** [sktime/skpro#934]
## Description

The implementation of `SquaredDistrLoss` appears to use `log_pdf`
instead of `pdf` when computing the squared distribution loss.

Current code:

res = -2 * y_pred.log_pdf(y_true) + y_pred.pdfnorm(a=2)

However the squared distribution loss is defined as:

L(y,d) = -2 * p_d(y) + ||p_d||²

where p_d(y) is the probability density function.

Using log_pdf changes both the magnitude and sign of the loss.

## Minimal example

```python
import numpy as np
import pandas as pd
from skpro.distributions.normal import Normal
from skpro.metrics._classes import SquaredDistrLoss

d = Normal(mu=[[0.0]], sigma=[[1.0]])
y_true = pd.DataFrame([[0.0]])

loss = SquaredDistrLoss()
print(loss._evaluate_by_index(y_true, d))

**Comment by Mahaveerjain-18** [comment]
Hi, I’d like to work on this issue.
